internal/parse/action/ast_builder.go

```diff
@@ -331,12 +332,15 @@ func (v *astBuilder) visitFn_arg_expr(ctx actgrammar.IFn_arg_exprContext) tree.E
 		expr := &tree.ExpressionFunction{
 			Inputs: make([]tree.Expression, len(ctx.AllFn_arg_expr())),
 		}
-		funcName := util.ExtractSQLName(ctx.Sfn_name().GetText())
-		f, ok := tree.SQLFunctions[strings.ToLower(funcName)]
+		expr.Function = util.ExtractSQLName(ctx.Sfn_name().GetText())
+
+		// this is a really ugly dependency; not quite circular, so it is ok.
+		// Since we likely will not make many changes to actions (in favor of procedures),
+		// it is ok for now
+		_, ok := engine.Functions[expr.Function]
```

Comment:
Yeah, it's coupled to what engine defines. We'll need to declare what's supported in this pkg, it's only a list of names

Reply:
hmm, wait, we used to have a function list in `tree` package

Reply:
Yeah I moved it

Reply:
Most of it was unused and it didn't include type info